lib/normalizer.c

```diff
@@ -849,7 +849,42 @@ grn_inline static const unsigned char *
 grn_nfkc_normalize_unify_alphabet_diacritical_mark(
   const unsigned char *utf8_char, unsigned char *unified)
 {
-  // TODO: Implement function to return `unified` instead of `utf8_char`.
+  /* U+00E0 LATIN SMALL LETTER A WITH GRAVE ..
+   * U+00E5 LATIN SMALL LETTER A WITH RING ABOVE
+   *
+   * U+0101 LATIN SMALL LETTER A WITH MACRON
+   * U+0103 LATIN SMALL LETTER A WITH BREVE
+   * U+0105 LATIN SMALL LETTER A WITH OGONEK
+   *
+   * U+01CE LATIN SMALL LETTER A WITH CARON
+   * U+01DF LATIN SMALL LETTER A WITH DIAERESIS AND MACRON
+   * U+01E1 LATIN SMALL LETTER A WITH DOT ABOVE AND MACRON
+   * U+01FB LATIN SMALL LETTER A WITH RING ABOVE AND ACUTE
+   *
+   * U+0201 LATIN SMALL LETTER A WITH DOUBLE GRAVE
+   * U+0203 LATIN SMALL LETTER A WITH INVERTED BREVE
+   * U+0227 LATIN SMALL LETTER A WITH DOT ABOVE
+   *
+   * U+1E01 LATIN SMALL LETTER A WITH RING BELOW,
+   * U+1EA1 LATIN SMALL LETTER A WITH DOT BELOW ..
+   * U+1EB7 LATIN SMALL LETTER A WITH BREVE AND DOT BELOW
+   */
+  if ((utf8_char[0] == 0xC3 && 0xA0 <= utf8_char[1] && utf8_char[1] <= 0xA5) ||
+      (utf8_char[0] == 0xC4 && (utf8_char[1] == 0x81 || utf8_char[1] == 0x83 ||
+                                utf8_char[1] == 0x85)) ||
```

Comment:
↓みたいに各条件の前にどれに対応するかコメントを置いてもらえますか？

```suggestion
  /*
   * U+00E0 LATIN SMALL LETTER A WITH GRAVE ..
   * U+00E5 LATIN SMALL LETTER A WITH RING ABOVE
   */
  if ((utf8_char[0] == 0xC3 && 0xA0 <= utf8_char[1] && utf8_char[1] <= 0xA5) ||
      /*
       * U+0101 LATIN SMALL LETTER A WITH MACRON
       * U+0103 LATIN SMALL LETTER A WITH BREVE
       * U+0105 LATIN SMALL LETTER A WITH OGONEK
       */ 
      (utf8_char[0] == 0xC4 && (utf8_char[1] == 0x81 || utf8_char[1] == 0x83 ||
                                utf8_char[1] == 0x85)) ||
```

Reply:
fix: [1e9137e](https://github.com/groonga/groonga/pull/1859/commits/1e9137e1ab47ccf5351f9baeddaba4659c394cf3) ブロッグごとに分けて入れていたのですが、条件と説明の距離が相手読みづらいので直前に追加するようにしました 🙏🏾